### ISSUE TITLE
Visual editor?

### ISSUE THREAD

**Issue by robertsdionne** [facebookarchive/caffe2#12]
I read the caffe2 source code and the "Improving Caffe: Some Refactoring" slides and was thinking that a visual model editor would be nice. The new operator/tensor separation reminds me of [PureData](https://puredata.info/), a dataflow programming language for digital audio signal processing, or the [Blender](https://www.blender.org/) Nodes interface, for 3D shader programming. I saw someone has hacked a caffe1 editor into the Blender Nodes interface here: [caffe-gui-tool](https://github.com/Chasvortex/caffe-gui-tool) and I have read through this caffe1 issue "GUI version of tools?": https://github.com/BVLC/caffe/issues/481

I went ahead and mocked up what the initialization and forward propagation phases of the caffe2 mnist model might look like in PureData (a task that would require adding tensor datatypes), though I did not finish any gradient or training ops:
![example_mnist](https://cloud.githubusercontent.com/assets/129368/9622833/7a2bfd50-5106-11e5-8f1a-8c57595df764.png)

Notably, PureData requires inputs to propagate from the top of the page to the bottom, which contradicts the standard depiction of neural networks processing data from the bottom up.

I think a visual editor might broaden the audience that can toy with these powerful models and encourage novel extensions.

**Comment by robertsdionne** [comment]
TensorFlow: https://twitter.com/graphific/status/663733506162483200

**Comment by Yangqing** [comment]
(Closing legasy issues... Actually, TensorBoard is a good visualizer front end to use.
